pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -452,6 +453,67 @@ func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance, opti
 	}(l, vmi)
 }
 
+func (l *LibvirtDomainManager) SetGuestTime(vmi *v1.VirtualMachineInstance) error {
+	// Try to set VM time to the current value.  This is typically useful
+	// when clock wasn't running on the VM for some time (e.g. during
+	// suspension or migration), especially if the time delay exceeds NTP
+	// tolerance.
+	// It is not guaranteed that the time is actually set (it depends on guest
+	// environment, especially QEMU agent presence) or that the set time is
+	// very precise (NTP in the guest should take care of it if needed).
+
+	domName := api.VMINamespaceKeyFunc(vmi)
+	dom, err := l.virConn.LookupDomainByName(domName)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("failed to sync guest time")
+		return err
+	}
+	// try setting the guest time for 5 seconds. Fail early if agent is not configured.
+	go l.setGuestTime(vmi, dom)
```

Comment:
How bad is this if we start doing this in parallel? Do you want to introduce a lock or something to avoid calling it multiple times? Less concerned about the guest, more about scalability in general.

Reply:
Maybe it is ok to not lock, but maybe we can just start fresh with a new ticker whenever this is called? I would want to avoid having too many parallel runs, while at the same time be sure that things like rapid pause/unpause are handled properly with an easy logic ...

Reply:
Yup, done, thanks